cli/src/main.rs

```diff
@@ -246,8 +248,28 @@ fn compile_once(world: &mut SystemWorld, command: &CompileSettings) -> StrResult
     match typst::compile(world) {
         // Export the PDF.
         Ok(document) => {
-            let buffer = typst::export::pdf(&document);
-            fs::write(&command.output, buffer).map_err(|_| "failed to write PDF file")?;
+            if command.output.extension().unwrap_or(OsStr::new("pdf")) == "png" {
+                let pixel_per_pt = 2.0;
+                let pixmaps: Vec<_> = document
+                    .pages
+                    .iter()
+                    .map(|frame| typst::export::render(frame, pixel_per_pt, Color::WHITE))
+                    .collect();
+
+                for (i, pixmap) in pixmaps.iter().enumerate() {
+                    let mut output = command.output.clone();
+                    output.set_file_name(format!(
+                        "{}_{}.png",
+                        command.output.file_stem().unwrap().to_str().unwrap(),
+                        i
+                    ));
+                    pixmap.save_png(&output).unwrap();
```

Comment:
Perhaps remaining consistent with the `fs::write` error below could be better
```suggestion
                    pixmap.save_png(&output).map_err(|_| "failed to write PNG file")?;
```

Reply:
makes sense, thanks!